core/src/code/errors.rs

```diff
@@ -132,6 +132,14 @@ pub enum ImportError {
     /// The signature of an imported function is invalid.
     #[display(fmt = "Invalid function signature for imported function with index `{_0}`")]
     InvalidImportFnSignature(u32),
+    /// Unexpected import kind.
+    #[display(fmt = "Unexpected import kind `{kind}` with index `{index}`")]
+    UnexpectedImportKind {
+        /// Kind of the import.
+        kind: &'static &'static str,
+        /// Index of the import.
+        index: u32,
+    },
```

Comment:
I cannot understand why `kind` has type `&'static &'static str`. If it's for optimisation purposes, than `&'static str` is also pointer and have the same size.

Reply:
https://github.com/gear-tech/gear/pull/3911#discussion_r1574580698

They'are not if I understand you correctly

Reply:
&'static str is a fat ptr, &'static &'static str is plain ptr

Reply:
Not sure if it's worth it for optimization purposes though.

Reply:
Yes, forgot about this. Ok